File: android/src/main/java/com/getcapacitor/community/classes/options/IsFingerprintValidOptions.java
package com.getcapacitor.community.classes.options;

import static com.getcapacitor.community.BridgefyHelper.makeUUID;

import android.util.Base64;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import java.util.UUID;

public class IsFingerprintValidOptions {

    @Nullable
    private UUID userID;

    @Nullable
    private byte[] fingerprint;

    public IsFingerprintValidOptions(PluginCall call) {
        @Nullable
        String userID = call.getString("userID");
        this.setUserID(userID);

        @Nullable
        String fingerprint = call.getString("fingerprint");
        this.setFingerprint(fingerprint);
    }

    private void setUserID(@Nullable String userID) {
        this.userID = makeUUID(userID);
    }

    private void setFingerprint(@Nullable String fingerprint) {
        this.fingerprint = (fingerprint == null || fingerprint.isEmpty()) ? null : Base64.decode(fingerprint, Base64.NO_WRAP);
    }

    @Nullable
    public UUID getUserID() {
        return userID;
    }

    @Nullable
    public byte[] getFingerprint() {
        return fingerprint;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Options/IsFingerprintValidOptions.swift
import Foundation
import Capacitor

@objc public class IsFingerprintValidOptions: NSObject {
    private var userID: UUID?
    private var fingerprint: Data?

    init(_ call: CAPPluginCall) {
        super.init()

        if let userID = call.getString("userID") {
            self.setUserID(userID)
        }
        if let fingerprint = call.getString("fingerprint") {
            self.setFingerprint(fingerprint)
        }
    }

    func setUserID(_ userID: String?) {
        self.userID = Helper.makeUUID(userID)
    }
    func setFingerprint(_ fingerprint: String?) {
        self.fingerprint = Helper.makeData(fingerprint)
    }

    func getUserID() -> UUID? {
        return self.userID
    }
    func getFingerprint() -> Data? {
        return self.fingerprint
    }
}
